tools/firely-server-import.rst

```diff
@@ -0,0 +1,182 @@
+.. _tool_fsi:
+
+Firely Server Import (FSI)
+==========================
+
+**Firely Server Import (FSI)** is a CLI application designed to optimize massive resource ingestion into a Firely Server instance. In contrast to resource ingestion by sending HTTP requests to a running Firely Server instance, this tool writes data directly to the underlying FS database which increases the throughput significantly.
+
+.. note::
+    Currently, this tool only supports import into SQL Server database, however, we are working on support for MongoDB.
+
+.. Installation
+.. ------------
+.. TBD: describe the usage of dotnet install command
+
+General usage
+-------------
+
+.. attention::
+
+  * All Firely Server instances targeting the same database must be stopped while the import is performed.
+  * Only one instance of FSI per database can be run at a time.
+
+Prerequisites
+^^^^^^^^^^^^^
+The tool requires that the target SQL Server database already exists and contains all required tables and indexes. If you don't have a database with the schema yet, you first need to run the Firely Server as described in article :ref:`configure_sql`.
+
+
+Input files formats
+^^^^^^^^^^^^^^^^^^^
+
+FSI supports the following input file format:
+
+* FHIR *collection* bundles stored in ``*.json`` files, and
+* ``*.ndjson`` files where each line contains a separate FHIR resource in JSON format.
+
+
+After the import
+^^^^^^^^^^^^^^^^
+
+After ingesting massive amount of data, it is important to make sure the SQL Server indexes are in good shape. You can read more on this topic here: :ref:`sql_index_maintenance`.
+
+Arguments
+---------
+
+The execution of FSI can be configured using input parameters. These parameters can be supplied either as CLI arguments or specified in the file ``appsettings.instance.json`` which must be created in the same directory as the ``fsi.exe``.
+
+If you want to specify input parameters in the file, you can use the snippet below as a base for your ``appsettings.instance.json``. In this case, you need to update the values that you want to set yourself and delete all other records.
+
+.. code-block:: JavaScript
+
+  {
+    "source": "./fsi-source", //valid directory
```

Comment:
Add somewhere that this directory is recursively visited.

Reply:
done